sanity.config.ts

```diff
@@ -14,10 +16,48 @@ export default defineConfig({
   plugins: [
     deskTool(),
     visionTool(),
+    // media(),
     //googleMapsInput(),
   ],
-
   schema: {
     types: schemaTypes,
+    templates: (prev, context) => {
+      const typesUsersCanCreate = ['movie']
+      const {currentUser} = context
+      const isAdmin = currentUser?.roles?.some((role) => role.name === 'administrator')
+      //if it's an admin, just give the regular functionality
+      if (isAdmin) {
+        return prev
+      }
+      const orgId = currentUser?.roles?.[0]?.name
+      //if they're in an org, filter out the "default" templates, so we ALWAYS
+      //create documents with our org id
+      const filteredTemplates = prev.filter((template) => {
+        return !typesUsersCanCreate.includes(template.id)
+      })
```

Comment:
Initial templates are NOT async. How can we search the Sanity dataset to get a "org" that is a document we can reference and insert in an initial value template?

Reply:
Top-level await (if people are cool with it / not using Safari)

Or wrap the studio in another app and then do something similar to top-level await

(up top -- do a client fetch, get all orgs, create a mapping of orgs to user role)